app/src/main/java/com/example/darknight/mi2016/LoginActivity.java

```diff
@@ -3,40 +3,141 @@
 
 import android.content.Intent;
 import android.net.Uri;
+import android.os.AsyncTask;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
 import com.google.android.gms.appindexing.Action;
 import com.google.android.gms.appindexing.AppIndex;
 import com.google.android.gms.appindexing.Thing;
 import com.google.android.gms.common.api.GoogleApiClient;
 
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.regex.Pattern;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
 public class LoginActivity extends AppCompatActivity {
     /**
      * ATTENTION: This was auto-generated to implement the App Indexing API.
      * See https://g.co/AppIndexing/AndroidStudio for more information.
      */
     private GoogleApiClient client;
+    Button login_button;
+    Button reg_later_button;
+    EditText mi_no;
+    Button submit_button;
+    JSONObject Jobject;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
-        final Button button = (Button) findViewById(R.id.register_later);
-        button.setOnClickListener(new View.OnClickListener() {
+
+        login_button = (Button) findViewById(R.id.mi_number);
+        reg_later_button = (Button) findViewById(R.id.register_later);
+        mi_no = (EditText) findViewById(R.id.mi_number_input);
+        submit_button = (Button) findViewById(R.id.login_submit);
+
+        login_button.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                login_button.setVisibility(View.GONE);
+                mi_no.setVisibility(View.VISIBLE);
+                submit_button.setVisibility(View.VISIBLE);
+            }
+        });
+
+        submit_button.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+
+                String mi_no_text = mi_no.getText().toString().toUpperCase();
+                Pattern pattern = Pattern.compile("MI-[A-Z]{3}-[0-9]{3,4}");
```

Comment:
Are you sure ki 000 would not be a valid MI number. Some people have special MI numbers such as 007, 001 etc. Some may have 000 at the end?